File: engine/glyphs.py
class GlyphLoader:
    def __init__(self):
        self.registry = {}

    def register(self, uri, func):
        self.registry[uri] = func

    def load(self, uri):
        if uri in self.registry:
            return self.registry[uri]()
        else:
            print(f"[WARN] Glyph not found: {uri}")

File: engine/glyphs.ts
// File: engine/glyphs.ts

type GlyphFactory = () => any;

export class GlyphLoader {
  private registry: Record<string, GlyphFactory> = {};

  register(uri: string, factory: GlyphFactory): void {
    this.registry[uri] = factory;
  }

  load(uri: string): any {
    const factory = this.registry[uri];
    if (factory) return factory();
    console.warn(`[WARN] Glyph not found: ${uri}`);
    return null;
  }
}
